homeassistant/components/vicare/climate.py

```diff
@@ -298,16 +299,38 @@ def set_preset_mode(self, preset_mode: str) -> None:
                 f"Cannot set invalid vicare program: {preset_mode}/{vicare_program}"
             )
 
-        _LOGGER.debug("Setting preset to %s / %s", preset_mode, vicare_program)
+        _LOGGER.debug("Current preset %s", self._current_program)
         if self._current_program != VICARE_PROGRAM_NORMAL:
             # We can't deactivate "normal"
+            _LOGGER.debug("deactivating %s", self._current_program)
+            current_program = ""
+            if self._current_program is not None:
+                current_program = self._current_program
```

Comment:
Suggestions to avoid this are welcome.

Reply:
See my suggestion but I would add a `None` check of `self._current_program` to `if self._current_program != VICARE_PROGRAM_NORMAL`